Dev/sealcid

hugoponthieu opened the pull request:
# Description

Sealci as a lot of component and therefore this is important to make this app launchable.
Let's create a deamon that is reachable with grpc that will manage each component of sealci.

hugoponthieu committed:
Init sealcid

Signed-off-by: hugoPonthieu <hugo.ponthieu@etu.umontpellier.fr>

hugoponthieu committed:
feat:lib structure

Signed-off-by: hugoPonthieu <hugo.ponthieu@etu.umontpellier.fr>

hugoponthieu committed:
feat:adding error enum

Signed-off-by: hugoPonthieu <hugo.ponthieu@etu.umontpellier.fr>

hugoponthieu committed:
feat(app):add trait & status

Signed-off-by: hugoPonthieu <hugo.ponthieu@etu.umontpellier.fr>

hugoponthieu committed:
fix:configure interface

Signed-off-by: hugoPonthieu <hugo.ponthieu@etu.umontpellier.fr>

hugoponthieu committed:
fine tune app trait

Signed-off-by: hugoPonthieu <hugo.ponthieu@etu.umontpellier.fr>

hugoponthieu committed:
feat:use workspsace to distribute traits

Signed-off-by: hugoPonthieu <hugo.ponthieu@etu.umontpellier.fr>

hugoponthieu committed:
feat:implement agent to app

Signed-off-by: hugoPonthieu <hugo.ponthieu@etu.umontpellier.fr>

hugoponthieu commented:
Little view of what should be implemented for the global config of the application:

![image](https://github.com/user-attachments/assets/13a581bc-6bca-4b10-82aa-ad50e974da16)

Also we should let the client choose if each services should be used. We should have a way to monitor each service that exist in  sealci.

hugoponthieu commented:
Component that need to be adapted for sealcid to work:
- [ ] Controller: create app struct and config
- [ ] scheduler: make the configuration public

hugoponthieu commented:
## Mutate 

We must implement for each component a way to mutate state of config and the process that is running. A mutate function should:

- mutate the global config
- create a config for the component that we want to change
- restart components that need to be the config mutation

Example:
```rust
pub struct Daemon {
	pub global_config: Arc<RwLock<GlobalConfig>>
	pub agent: Arc<RwLock<AgentApp>>
}

impl Daemon {
	fn mutate_agent(&self, new_confg: Config) {
		let global_config = global_config.write().await;
		// perform modification on global config

		// Then build the config for agent
		let agent_config= global_config.into();
		let agent = agent.write().await;
		agent= AgentApp::configure();
		agent.start();	
	}
	
}
```

hugoponthieu committed:
feat:daemon, global config, app mutation

Signed-off-by: hugoPonthieu <hugo.ponthieu@etu.umontpellier.fr>

hugoponthieu committed:
feat:placeholder function mutation

Signed-off-by: hugoPonthieu <hugo.ponthieu@etu.umontpellier.fr>

hugoponthieu committed:
Merge pull request #129 from dev-sys-do/feat/sealcid-global-config

Feat/sealcid global config

theotchlx commented:
[PR 127](https://github.com/dev-sys-do/sealci/pull/127) (2^7 - 1) implements the struct trait as a separate package with its own Cargo.toml, making it importable and implementable by other sealci components, avoiding circular dependency issue (if it were defined in sealcid)

Have you found another solution in the meantime?

theotchlx committed:
fix(scheduler): test imports and upgrade dependencies

Signed-off-by: Théo Tchilinguirian <theo.tchlx@proton.me>

hugoponthieu committed:
feat:toggle services

Signed-off-by: hugoPonthieu <hugo.ponthieu@etu.umontpellier.fr>

theotchlx committed:
chore(scheduler/lib): resolve warnings, deprecated methods

Signed-off-by: Théo Tchilinguirian <theo.tchlx@proton.me>

theotchlx committed:
chore(scheduler/app): remove commented code

Signed-off-by: Théo Tchilinguirian <theo.tchlx@proton.me>

hugoponthieu committed:
feat:import proto

Signed-off-by: hugoPonthieu <hugo.ponthieu@etu.umontpellier.fr>

hugoponthieu committed:
feat:impl grpc for deamon

Signed-off-by: hugoPonthieu <hugo.ponthieu@etu.umontpellier.fr>

duratm committed:
feat(sealcid/controller): add controller implementation for sealcid

Signed-off-by: duratm <duratmathias@gmail.com>

# Conflicts:
#	sealcid/Cargo.lock
#	sealcid/Cargo.toml
#	sealcid/src/lib/common/error.rs

duratm committed:
feat(sealcid/monitor): add monitor implementation for sealcid

Signed-off-by: duratm <duratmathias@gmail.com>

hugoponthieu committed:
feat:grpc server

Signed-off-by: hugoPonthieu <hugo.ponthieu@etu.umontpellier.fr>

hugoponthieu committed:
Merge pull request #132 from dev-sys-do/feat/sealcid-proto-client-server

Feat/sealcid proto client server

duratm committed:
feat(sealcid/grpc): fix implementations with grpc and update global config + rebase

Signed-off-by: duratm <duratmathias@gmail.com>

duratm committed:
feat(sealcid/grpc): fix names and format

Signed-off-by: duratm <duratmathias@gmail.com>

Courtcircuits commented:
I'll be implementing the release agent part of sealcid :smile:

duratm committed:
Merge pull request #131 from dev-sys-do/dev/sealcid-controller

feat(sealcid/controller): add controller implementation for sealcid

theotchlx committed:
feat(scheduler/server): return ActionStatus Error on unavailable Agent

Signed-off-by: Théo Tchilinguirian <theo.tchlx@proton.me>

hugoponthieu committed:
feat:client cli

Signed-off-by: hugoPonthieu <hugo.ponthieu@etu.umontpellier.fr>

hugoponthieu committed:
wip:fix controller mutation

Signed-off-by: hugoPonthieu <hugo.ponthieu@etu.umontpellier.fr>

theotchlx committed:
feat(scheduler): massively improve error handling.

Signed-off-by: Théo Tchilinguirian <theo.tchlx@proton.me>

duratm committed:
feat(sealcid/grpc): fix config in progress + add scheduler in sealcid

Signed-off-by: duratm <duratmathias@gmail.com>

duratm committed:
feat(sealcid/grpc): remove unused mutation

Signed-off-by: duratm <duratmathias@gmail.com>

hugoponthieu committed:
Merge pull request #135 from dev-sys-do/dev/sealcid-scheduler

Add scheduler in sealcid

hugoponthieu committed:
fix:daemon

Signed-off-by: hugoPonthieu <hugo.ponthieu@etu.umontpellier.fr>

hugoponthieu committed:
Merge pull request #136 from dev-sys-do/feat/sealcid-client

Feat/sealcid client

hugoponthieu committed:
fix:controller error handling

Signed-off-by: hugoPonthieu <hugo.ponthieu@etu.umontpellier.fr>

hugoponthieu committed:
Merge branch 'main' into dev/sealcid

theotchlx committed:
Merge branch 'dev/scheduler' into dev/sealcid

hugoponthieu committed:
launch daemon grpc socket

Signed-off-by: hugoPonthieu <hugo.ponthieu@etu.umontpellier.fr>

theotchlx committed:
feat(scheduler): implement sealcid integration

Signed-off-by: Théo Tchilinguirian <theo.tchlx@proton.me>

theotchlx committed:
feat(scheduler): move logger init from lib to main

Signed-off-by: Théo Tchilinguirian <theo.tchlx@proton.me>

hugoponthieu committed:
fix:deadlock scheduler

Signed-off-by: hugoPonthieu <hugo.ponthieu@etu.umontpellier.fr>

theotchlx committed:
feat(scheduler): improve logs

Signed-off-by: Théo Tchilinguirian <theo.tchlx@proton.me>

theotchlx committed:
docs(sealcid): write architecture doc

Signed-off-by: Théo Tchilinguirian <theo.tchlx@proton.me>

hugoponthieu committed:
fix:removing print

Signed-off-by: hugoPonthieu <hugo.ponthieu@etu.umontpellier.fr>

hugoponthieu committed:
fix:starting monitor

Signed-off-by: hugoPonthieu <hugo.ponthieu@etu.umontpellier.fr>

duratm committed:
feat(toggle+retry): add retry on grpc instantiation on agent + add toggle managment in sealcid mutations + remove the mutation file again

Signed-off-by: duratm <duratmathias@gmail.com>

hugoponthieu committed:
fix:client default config & scheduler logging

Signed-off-by: hugoPonthieu <hugo.ponthieu@etu.umontpellier.fr>

hugoponthieu committed:
feat: improve monitor listening logs

Signed-off-by: hugoPonthieu <hugo.ponthieu@etu.umontpellier.fr>

duratm committed:
feat(toggle+retry): fix deadlock between mutex read and write

Signed-off-by: duratm <duratmathias@gmail.com>

hugoponthieu committed:
Merge pull request #140 from dev-sys-do/dev/sealcid-retry-toggle

feat(toggle+retry): add retry on grpc instantiation on agent + add to…

hugoponthieu committed:
fix:monitor controller communication

Signed-off-by: hugoPonthieu <hugo.ponthieu@etu.umontpellier.fr>

hugoponthieu committed:
feat:improve logging

Signed-off-by: hugoPonthieu <hugo.ponthieu@etu.umontpellier.fr>

hugoponthieu committed:
Merge pull request #141 from dev-sys-do/feat/sealcid-grpc-sever

Feat/sealcid grpc sever

hugoponthieu committed:
fix:controller actix data

Signed-off-by: hugoPonthieu <hugo.ponthieu@etu.umontpellier.fr>

hugoponthieu committed:
fix:agent default config

Signed-off-by: hugoPonthieu <hugo.ponthieu@etu.umontpellier.fr>

theotchlx committed:
Merge remote-tracking branch 'origin/main' into dev/sealcid

theotchlx committed:
feat(scheduler/config): switch default addr to IPv6

Signed-off-by: Théo Tchilinguirian <theo.tchlx@proton.me>

theotchlx committed:
feat(scheduler/errors): clean errors enum

Signed-off-by: Théo Tchilinguirian <theo.tchlx@proton.me>

theotchlx committed:
chore(scheduler/agent_client): add comments

Signed-off-by: Théo Tchilinguirian <theo.tchlx@proton.me>

theotchlx committed:
feat(scheduler/agent_server): improve logging

Signed-off-by: Théo Tchilinguirian <theo.tchlx@proton.me>

theotchlx committed:
docs(scheduler/readme): update readme

Signed-off-by: Théo Tchilinguirian <theo.tchlx@proton.me>

duratm committed:
feat(toggle+retry): implement status display for the services

Change status to use the proto ones
Add status into the commands

Signed-off-by: duratm <duratmathias@gmail.com>

duratm committed:
Merge pull request #142 from dev-sys-do/dev/sealcid-status

Dev/sealcid status

hugoponthieu committed:
feat:toggle & process

Signed-off-by: hugoPonthieu <hugo.ponthieu@etu.umontpellier.fr>

# Conflicts:
#	monitor/src/app/mod.rs
#	sealcid/src/lib/server/grpc.rs

duratm committed:
feat(toggle+retry): change enable and disable function and fix monitor shutdown

Signed-off-by: duratm <duratmathias@gmail.com>

duratm committed:
feat(toggle+retry): add server_handle to controller + change println to info

Signed-off-by: duratm <duratmathias@gmail.com>

duratm committed:
feat(toggle+retry): add start and stop capaylities

Signed-off-by: duratm <duratmathias@gmail.com>

Courtcircuits committed:
feat: added dns option to define dns nameserver

Signed-off-by: Courtcircuits <tristan-mihai.radulescu@etu.umontpellier.fr>

theotchlx commented:
This is great work :)

Services implement the fundamental properties of the App trait, sealcid works for deploying 1 of each service.

Scheduler implementation branch follows sealcid development, and has also been improved on error handling (among other things) and should now allow Controller to handle Agent death.

Courtcircuits committed:
fix: check for /etc

Signed-off-by: Courtcircuits <tristan-mihai.radulescu@etu.umontpellier.fr>

Courtcircuits committed:
Merge pull request #144 from dev-sys-do/feat/compactor-dns

feat: added dns option to define dns nameserver

theotchlx commented:
Last improvements for the Scheduler :')!

duratm committed:
feat(sealcid/r-a): wip running but blocking and bugging

Signed-off-by: duratm <duratmathias@gmail.com>

duratm committed:
Merge pull request #143 from dev-sys-do/feat/toggle-process

feat:toggle & process

duratm committed:
feat(sealcid/r-a): reelase agent working

raf:
- handle shutdown
- make start not crash

Signed-off-by: duratm <duratmathias@gmail.com>

duratm committed:
Merge pull request #151 from dev-sys-do/feat/release-agent-sealcid

feat(sealcid/r-a): wip running but blocking and bugging

hugoponthieu committed:
Merge remote-tracking branch 'origin/main' into dev/sealcid

hugoponthieu committed:
feat:register route for release agent

Signed-off-by: hugoPonthieu <hugo.ponthieu@etu.umontpellier.fr>

hugoponthieu committed:
fix:dumplet

Signed-off-by: hugoPonthieu <hugo.ponthieu@etu.umontpellier.fr>

theotchlx commented:
Yipee! Good job!! :-)
